internal/harnesses/remote/k8s.go

```diff
@@ -0,0 +1,382 @@
+package remote
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/url"
+	"slices"
+
+	"github.com/chainguard-dev/terraform-provider-imagetest/internal/harnesses/base"
+	"github.com/chainguard-dev/terraform-provider-imagetest/internal/types"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/httpstream"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+const (
+	ImagetestNamespaceNameFormat  = "imagetest-%s"
+	ImagetestSandboxPodNameFormat = "%s-sandbox"
+	EntrypointContainerName       = "wolfi-base"
+	RbacResourcesNameFormat       = "rbac-%s"
+)
+
+// Ensure type k8s conforms to types.Harness.
+var _ types.Harness = &k8s{}
+
+type k8s struct {
+	// Provide basic functions needed to make the harness operate.
+	*base.Base
+
+	// ID to identify resources created by this harness.
+	Id string
+
+	// RESTClient config.
+	Config *rest.Config
+
+	// Kubernetes Client instance.
+	Client kubernetes.Interface
+}
+
+func New(id string, kubeconfig *string) (types.Harness, error) {
```

Comment:
we'd save some future refactoring if we start with variadic options here, can we optionally load the kubeconfig via `WithKubeconfig()`?

Reply:
By variadic args do you mean `... interface{}` or just `... *string`?

Reply:
Or do you want a builder-style option configuration?

Reply:
re-reading through this, its fine to add this refactor after the bulk of this lands 🙏

Reply:
Cool, sounds good. I still have to add some tests though and the change wouldn't be all that complicated.